### PR TITLE
ENH: always use zip64, upgrade pickle protocol to 3

### LINKED ISSUE #10776: savez fails on large array of objects
I'm getting a RuntimeError "File size unexpectedly exceeded ZIP64 limit" when using `savez` to save a few arrays to a file, the largest of which is an array of matrices each with a different number of rows (so that the top level array's dtype is object). As best I can tell, the condition used by numpy internally to decide whether or not to pass `force_zip64=True` to the zipfile open is checking if any of the arrays to be put into the archive has nbytes greater than 2^30. My array of matrices reports nbytes less than 2^30 but in reality the total number of bytes exceeds this (it's about 1.7GB). I'm using python 3.6 and numpy 1.14.2 on Ubuntu 16.04. 

The following code produces the error:
```python
test_data = np.asarray([np.random.rand(np.random.randint(50,100),4) for i in range(800000)])
np.savez('test', test_data=test_data)
```
whereas changing `np.random.randint(50,100)`->`75` produces no error.

### PR DESCRIPTION
**Pull request by mattip**
Fixes #10776 but forces use of pickle protocol 3 for writing arrays with objects, otherwise the new test fails with `MemoryError`

This will break backward compatibility with python2, so pushing it off to 1.17

Edit: add explanation why protocol 3 is needed

### CODE REVIEW

**Comment by mattip** [comment]
rebased off master to try to clear the macos build error

**Comment by mattip** [comment]
ping

**Comment by charris** [comment]
Needs rebase.

**Comment by mattip** [comment]
rebased, squashed, tests are passing

File: numpy/lib/tests/test_io.py [hunk 1]
```diff
@@ -574,6 +572,20 @@ def test_unicode_and_bytes_fmt(self, fmt, iotype):
         else:
             assert_equal(s.read(), b"%f\n" % 1.)
 
+    @pytest.mark.skipif(sys.platform=='win32',
+                        reason="large files cause problems")
```
Comment (charris) [inline numpy/lib/tests/test_io.py]: Is this true of all Windows versions?

**Comment by charris** [comment]
We may want to increase the protocol level at some point. In the long run, protocol 5 is probably preferable for big files.

**Comment by charris** [comment]
Let's give this a shot.

**Comment by janTOffermann** [comment]
Is there any particular reason why the pickle protocol is not user-adjustable? (if I'm understanding things correctly, it looks like I might need to practically use protocol 4 for a particular task involving large files, so while this update is welcome it still doesn't solve this issue.)

**Pull request merged**